File: src/main/java/com/xiaolouJava/sql/GenericLoadSave.java
package com.xiaolouJava.sql;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.sql.DataFrame;
import org.apache.spark.sql.SQLContext;

/**
 * 通用的load和save操作
 */
public class GenericLoadSave {
    public static void main(String[] args) {
        SparkConf conf = new SparkConf().setMaster("local").setAppName("DataFrameCreate");
        JavaSparkContext sc = new JavaSparkContext(conf);
        SQLContext sqlContext = new SQLContext(sc);

        DataFrame usersDF = sqlContext.read().load("D://BigData//JetBrains//sparktest//src//txts//users.parquet");
//        usersDF.printSchema();
//        usersDF.show();

        usersDF.select("name","favorite_color").write()
                .save("C://Users//33063//Desktop//amesAndColors.parquet");


    }
}

File: src/main/scala/com/xiaolou123/sql/GenericLoadSave.scala
package com.xiaolou123.sql

import org.apache.spark.{SparkConf, SparkContext}
import org.apache.spark.sql.SQLContext

/**
  * 通用的load和save操作
  */
object GenericLoadSave {

  def main(args: Array[String]): Unit = {
    val conf = new SparkConf().setMaster("local").setAppName("GenericLoadSave")
    val sc = new SparkContext(conf)
    val sqlContext = new SQLContext(sc)

    val usersDF = sqlContext.read.load("D://BigData//JetBrains//sparktest//src//txts//users.parquet")
    usersDF.write.save("C://Users//33063//Desktop//namesAndColors_scala.parquet")

  }
}
